conjure-runtime/src/client_factory.rs

```diff
@@ -301,16 +303,27 @@ impl ClientFactory {
             let service = service.to_string();
             move |c| c.merged_service(&service).unwrap_or_default()
         });
-
         let state_builder = self.state_builder(service);
+        Self::raw_client(state_builder, service_config, None)
+    }
 
-        let state = state_builder.build(&service_config.get(), None)?;
+    fn raw_client<T>(
+        state_builder: T,
+        service_config: Refreshable<ServiceConfig, Error>,
+        override_host_index: Option<usize>,
```

Comment:
I see this is part of the `CachedConfig` type. It says it `Overrides the `hostIndex` field included in metrics.` What is the purpose of that?

Reply:
There are some metrics which are tracked for each URI separately with a `hostIndex` field. Normally that's determined by the index of the URI in the URIs list of the ServiceConfig, but the per-host logic ends up making a separate ServiceConfig for each URI. Without the override each of those would end up using hostIndex 0 so the metrics would all get merged.